test/unit/gateways/nmi_test.rb

```diff
@@ -30,6 +31,66 @@ def setup
     }
   end
 
+  def test_successful_authorize_and_capture_using_security_key
```

Comment:
💬  Comment:

Again you are just updating the authentication method, why it should have an impact on the behavior of authorize, purchase or capture ?

And if it was the case why testing that remote behavior on the unit tests, unit tests are the place to check how the request is being created and how the response is formatted, Authentication only changes the request structure.

Reply:
I have added this type of test because when we are using security_key for some reason we need to use the test card number provided by the [documentation](https://secure.networkmerchants.com/gw/merchants/resources/integration/integration_portal.php#testing_information).